### PR TITLE
Switch configs in settings UI

### PR DESCRIPTION
**Pull request by rich-bugsnag**
## Goal

add switch specific config values to the config UI

Screenshot 2022-09-20 at 11 33 20

### CODE REVIEW

File: src/Assets/Bugsnag/Scripts/BugsnagSettingsObject.cs [hunk 1]
```diff
@@ -40,6 +40,12 @@ public class BugsnagSettingsObject : ScriptableObject
         public double SecondsPerUniqueLog = 5;
         public List<TelemetryType> Telemetry = new List<TelemetryType> { TelemetryType.InternalErrors };
         public int VersionCode;
+
+        public SwitchCacheType SwitchCacheType = SwitchCacheType.R;
+        public string SwitchCacheMountName = "BugsnagCache";
+        public int SwitchCacheIndex = 0;
+        public int SwitchMaxCacheSize = 10485760;
```
Comment (tomlongridge) [inline src/Assets/Bugsnag/Scripts/BugsnagSettingsObject.cs:47]: Suggest we rename this for consistency - perhaps that requires the underlying config option to change too? But feels worth it to save confusion.

```suggestion
        public int SwitchCacheMaxSize = 10485760;
```

File: src/Assets/Bugsnag/Editor/BugsnagEditor.cs [hunk 2]
```diff
@@ -105,6 +105,14 @@ private void DrawSettingsEditorWindow()
             {
                 DrawEndpoints(so);
             }
+
+            GUILayout.Space(5);
+            _showSwitch = EditorGUILayout.Foldout(_showSwitch, "Nintendo Switch (Requires Nintendo Switch Bugsnag plugin)", true);
```
Comment (gareththackeray) [inline src/Assets/Bugsnag/Editor/BugsnagEditor.cs:110]: I imagined this in a tooltip is that not possible on a group header?

Comment (rich-bugsnag) [reply src/Assets/Bugsnag/Editor/BugsnagEditor.cs]: Done 👍

File: src/Assets/Bugsnag/Scripts/BugsnagSettingsObject.cs (hunk 1 above)
Comment (rich-bugsnag) [reply src/Assets/Bugsnag/Scripts/BugsnagSettingsObject.cs]: Done

**Review by gareththackeray — APPROVED** [review]

**Pull request merged**